src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1542,18 +1655,17 @@ static UCS_CLASS_INIT_FUNC(uct_dc_mlx5_iface_t, uct_md_h tl_md, uct_worker_h wor
 
     self->tx.dci_pool_release_bitmap = 0;
 
+    self->tx.num_dci_pools = 0;
     if (ucs_test_all_flags(md->flags, UCT_IB_MLX5_MD_FLAG_DEVX_DCI |
                                       UCT_IB_MLX5_MD_FLAG_CQE_V1)) {
         self->flags           |= UCT_DC_MLX5_IFACE_FLAG_UIDX;
-        self->tx.num_dci_pools = self->super.super.super.num_paths;
     }
 
     if ((params->field_mask & UCT_IFACE_PARAM_FIELD_FEATURES) &&
         !(params->features & UCT_IFACE_FEATURE_PUT)) {
         self->flags |= UCT_DC_MLX5_IFACE_FLAG_DISABLE_PUT;
     }
-
-    max_dcis = ucs_min(INT8_MAX, UINT8_MAX / self->tx.num_dci_pools);
+    max_dcis = ucs_min(INT8_MAX, UINT8_MAX / UCT_DC_MLX5_IFACE_MAX_DCI_POOLS);
```

Comment:
revert?

Reply:
This will cause a division by 0 if we don't create at least 1 dci pool: https://github.com/openucx/ucx/pull/9665#discussion_r1590279124

Reply:
i think the check in line 1669 is not relevant anymore since the number of pools is dynamic